common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -193,16 +195,26 @@ def index_dictionary(self):
         """
         Return dictionary prepared with module content and type for indexing.
         """
-        result = super(CapaDescriptor, self).index_dictionary()
-        if not result:
-            result = {}
-        index = {
-            'content_type': self.INDEX_CONTENT_TYPE,
-            'problem_types': list(self.problem_types),
-            "display_name": self.display_name
+        xblock_body = super(CapaDescriptor, self).index_dictionary()
+        # Removing solution
+        capa_content = re.sub(re.compile(r"<solution>.*</solution>", re.DOTALL), "", self.data)
+        # Removing HTML-encoded non-breaking space characters
+        capa_content = re.sub(r"(\s|&nbsp;|//)+", " ", html_to_text(capa_content))
+        # Removing HTML CDATA
+        capa_content = re.sub(r"<!\[CDATA\[.*\]\]>", "", capa_content)
+        # Removing HTML comments
+        capa_content = re.sub(r"<!--.*-->", "", capa_content)
```

Comment:
`html_to_text` eliminates tags, which should only leave content visible to the user.

Additionally, I've removed html comments, CDATA and html non-breaking space characters.